src/Mobs/Path.h

```diff
@@ -0,0 +1,167 @@
+#pragma once
+
+#ifdef COMPILING_PATHFIND_DEBUGGER
+/* Note: the COMPILING_PATHFIND_DEBUGGER flag is used by Native/WiseOldMan95 to debug this class outside of MCServer.
+This preprocessor flag is never set when compiling MCServer. */
+#include "PathFinderIrrlicht_Head.h"
+#endif
+
+#include <vector>
+#include <queue>
+#include <unordered_map>
+
+/* MCServer forward declarations */
+#ifndef COMPILING_PATHFIND_DEBUGGER
+
+// fwd: Vector3.h
+template <typename T> class Vector3;
+typedef Vector3<double> Vector3d;
+
+// fwd: cChunkMap.h
+typedef cItemCallback<cChunk> cChunkCallback;
+#endif
+
+/* Various little structs and classes */
+enum class ePathFinderStatus {CALCULATING,  PATH_FOUND,  PATH_NOT_FOUND};
+struct cPathCell;  // Defined inside Path.cpp
+class compareHeuristics
+{
+public:
+	bool operator()(cPathCell * & a_V1,  cPathCell * &  a_V2);
```

Comment:
I'm trying to ditch pointers all together in the SmartPointers branch, not going well as of now.